iceoryx_examples/waitset/README.md

```diff
@@ -1,6 +1,566 @@
-# waitset - Callbacks
+# WaitSet
 
-**Hint** This example shows a work in progress and is not yet working.
+The WaitSet is a set where you can attach Trigger to signal a wide variety
+of events to one single listener. The typical approach is that one creates a
+WaitSet attaches multiple subscriber or other Trigger to it and then wait till
+one or many of the attach classes signal an event. If that happens one receives
+a list of all the Triggers which were triggered and the program can act accordingly.
 
-## Introduction
+The WaitSet is state based which means that it will trigger until the state which
+caused the trigger changed. For instance, when the subscriber reports to the WaitSet
+that a new sample has been received, the WaitSet will notify the user until the
+sample has been taken and released.
 
+## Glossary
+
+ - **Listener** a class which can be triggered by a _Trigger_, for instance a _WaitSet_.
```

Comment:
DDS also uses the term _Listener_, see [here](https://community.rti.com/static/documentation/connext-dds/5.2.0/doc/manuals/connext_dds/html_files/RTI_ConnextDDS_CoreLibraries_UsersManual/index.htm#UsersManual/Types_of_Listeners.htm#entities_943231811_994265%3FTocPath%3DPart%25202%253A%2520Core%2520Concepts%7CDDS%2520Entities%7CListeners%7C_____1]). Is it the same in our context or something different?

Reply:
The idea is the same but it is not a base class in our context. A listener listens to triggers or more in general it listens to events. Every time a specific event happens a trigger is activated and the listener will "hear" that.

Reply:
Renamed it to Notifyable.